### PR TITLE
Remove statusmanager

### LINKED ISSUE #3855: Remove statusmanager
# Feature Request
Remove statusmanager datasource endpoint type

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Description

- remove support for statusmanager endpoint type
- rename status_manager_url config param to shareServiceUrl

## Related issues or pull requests

closes #3855 

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: projects/hslayers/src/components/save-map/form/form.component.ts [hunk 1]
```diff
@@ -128,16 +128,13 @@ export class HsSaveMapAdvancedFormComponent implements OnDestroy, OnInit {
   /**
    * Check if user is allowed to save the composition, based on the currently selected endpoint type
    * @returns True if endpoint type is 'layman'.
-   * True if the endpoint type is 'statusmanager' and the user is authorized,
-   * false otherwise
+   * True if the user is authorized
```
Comment (jmacura) [inline projects/hslayers/src/components/save-map/form/form.component.ts:131]:
```suggestion
```

Actually the method only checks for "laymanity" of the current endpoint and for authorization

File: projects/test-app/src/hslayers-app/hslayers-app.component.ts [hunk 2]
```diff
@@ -436,7 +436,7 @@ export class HslayersAppComponent {
         {name: 'warning', url: '/assets/icons/warning.svg'},
         {name: 'wifi', url: '/assets/icons/wifi8.svg'},
       ],
-      status_manager_url: 'http://localhost:8086',
+      share_service_url: 'http://localhost:8086',
```
Comment (jmacura) [inline projects/test-app/src/hslayers-app/hslayers-app.component.ts:439]:
```suggestion
      shareServiceUrl: 'http://localhost:8086',
```

Let's go camelCase when we are changing it already (applies also to other occurences). What do you think?

**Review by jmacura — COMMENTED** [review]
So far I didn't test but seems OK to me.

File: projects/hslayers/src/components/compositions/compositions-catalogue.service.ts [hunk 3]
```diff
@@ -74,7 +74,7 @@ export class HsCompositionsCatalogueService {
     private _zone: NgZone
   ) {
     this.hsCommonEndpointsService.endpointsFilled.subscribe((endpoints) => {
-      this.filteredEndpoints = this.getFilteredEndpointsForCompositions();
+      this.endpoints = endpoints;
```
Comment (jmacura) [inline projects/hslayers/src/components/compositions/compositions-catalogue.service.ts:77]:
```suggestion
      this.endpoints = endpoints ?? [];
```

Better safe then sorry. `endpoints` could be `null` in case the `endpointsFilled` is never casted, since it's a BehaviorSubject initiated with a `null` value.

**Review by jmacura — COMMENTED** [review]
One more thing...

File: projects/hslayers/src/components/compositions/endpoints/compositions-micka.service.ts [hunk 4]
```diff
@@ -95,6 +98,7 @@ export class HsCompositionsMickaService {
    
    */
   compositionsReceived(endpoint: HsEndpoint, response: any): void {
+    console.warn(endpoint);
```
[the remark this answers is not in the document]
Comment (FilipLeitner) [reply projects/hslayers/src/components/compositions/endpoints/compositions-micka.service.ts:101]: Will remove

File: projects/hslayers/src/components/compositions/endpoints/compositions-micka.service.ts [hunk 5]
```diff
@@ -223,4 +229,66 @@ export class HsCompositionsMickaService {
...
+      return compData.url;
+    }
+    return compData.map((link) =>
+      typeof link == 'object' && link.url !== undefined ? link.url : link
+    );
+  }
+
+  delete(e: HsEndpoint, c: HsMapCompositionDescriptor) {
+    console.warn('Delete method for Micka compositions not implemented', e, c);
+  }
```
[the remark this answers is not in the document]
Comment (FilipLeitner) [reply projects/hslayers/src/components/compositions/endpoints/compositions-micka.service.ts:293]: Hmm can cant remember what I ment by this.  But thinking about this now I guess user is not supposed to remove micka records directly, right @fzadrazil ?

File: projects/test-app/src/hslayers-app/hslayers-app.component.ts (hunk 2 above)
Comment (FilipLeitner) [reply projects/test-app/src/hslayers-app/hslayers-app.component.ts:439]: Right, thats a great idea!

File: projects/hslayers/src/components/compositions/endpoints/compositions-micka.service.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (fzadrazil) [reply projects/hslayers/src/components/compositions/endpoints/compositions-micka.service.ts:288]: There are two cases:

1. Layman records with metadata in Micka - HSLayers should primarily display Layman record, not Micka, once the Layman record is deleted, it is automatically deleted also in Micka (Layman takes care of that)
2. Micka records with no link to Layman - user should not delete those in HSLayers

In case of some error when Layman record is removed but not Micka record, it is still responsibility of Layman to remove this, not HSLayers.

[the remark this answers is not in the document]
Comment (jmacura) [reply projects/hslayers/src/components/compositions/endpoints/compositions-micka.service.ts:288]: Deleting Micka records from HSLayers is already prevented in the UI. By setting `editable = false` property, the button is hidden here https://github.com/hslayers/hslayers-ng/blob/32610f066dfea25b8cecfff1760c3dd15de98ac9/projects/hslayers/src/components/compositions/compositions-list-item.component.html#L20 and further prevented here https://github.com/hslayers/hslayers-ng/blob/32610f066dfea25b8cecfff1760c3dd15de98ac9/projects/hslayers/src/components/compositions/compositions-list-item.component.ts#L96

So the code is fine with me like this. I propose to add some comment to the delete() method itself to make it clear for future ourselves, why it is not implemented 🙂

**Comment by jmacura** [comment]
For future reference, I quote the discussion about Micka's `delete()` method from the review comments here:

> Is this a kind of TODO for us or is it a Micka characteristics? In other words, shall it be implemented on our side?

---

> Hmm can cant remember what I ment by this. But thinking about this now I guess user is not supposed to remove micka records directly, right @fzadrazil ?

---

> There are two cases:
> 
>     1. Layman records with metadata in Micka - HSLayers should primarily display Layman record, not Micka, once the Layman record is deleted, it is automatically deleted also in Micka (Layman takes care of that)
> 
>     2. Micka records with no link to Layman - user should not delete those in HSLayers
> 
> 
> In case of some error when Layman record is removed but not Micka record, it is still responsibility of Layman to remove this, not HSLayers.

---

> Deleting Micka records from HSLayers is already prevented in the UI. By setting `editable = false` property, the button is hidden here
> 
> https://github.com/hslayers/hslayers-ng/blob/32610f066dfea25b8cecfff1760c3dd15de98ac9/projects/hslayers/src/components/compositions/compositions-list-item.component.html#L20
> and further prevented here
> 
> https://github.com/hslayers/hslayers-ng/blob/32610f066dfea25b8cecfff1760c3dd15de98ac9/projects/hslayers/src/components/compositions/compositions-list-item.component.ts#L96
> 
> So the code is fine with me like this. I propose to add some comment to the delete() method itself to make it clear for future ourselves, why it is not implemented 🙂

File: projects/hslayers/src/components/core/core.service.ts [hunk 6]
```diff
@@ -142,6 +142,7 @@ export class HsCoreService {
   }
 
   /**
+   * FIXME: statusmanager
```
Comment (jmacura) [inline projects/hslayers/src/components/core/core.service.ts:145]: What about this FIXME, @FilipLeitner ? Leave it for another PR?

Comment (FilipLeitner) [reply projects/hslayers/src/components/core/core.service.ts:145]: Seems like its not used at all and should be just deleted. Probably I left note there just to give it one more thought...but seems preety straight forward to me now. What you think?

Comment (jmacura) [reply projects/hslayers/src/components/core/core.service.ts:145]: No usage of `isAuthorized()` in our codebase any more. Seems safe to delete to me 👍

Comment (FilipLeitner) [reply projects/hslayers/src/components/core/core.service.ts]: [c24832d](https://github.com/hslayers/hslayers-ng/pull/4014/commits/c24832d71d768eaeef6f6e7847829556a7539fc3)

File: projects/hslayers/src/components/core/core.service.ts
Comment (jmacura) [inline projects/hslayers/src/components/core/core.service.ts:1]: Could you revert the Prettier-generated noise in this file, please?

**Review by jmacura — CHANGES_REQUESTED** [review]

**Review by jmacura — APPROVED** [review]
